### ISSUE TITLE
Mining Pool Hub Odd Behaviour

### ISSUE THREAD

**Issue by titvs** [MultiPoolMiner/MultiPoolMiner#1799]
Sorry to keep bugging you guys but it seems I found yet another bug to be bugged. ;p

On the latest commit, I’ve noticed some odd behaviour when running MPM configured with Mining Pool Hub (MPH) and Zergpool/Blockmasters. The counter for Pool Prices out of sync continues to increase and it does not switch algo on MPH.
So i started to troubleshoot to see if it was somekind of problem with one of the pools. I got the following results (keep in mind that all the other pools where configured on the excludepools switch):

- MPH and ZergPool - out of sync message and no algo switching
- MPH and Blockmasters - same problem
- MPH solo - no warning message
- ZergPool or Blockmasters solo - no warning message
- ZergPool and Blockmasters - no warning message

As you guys can see it seems to be some problem related to MPH ... I can attach the logs from the runs of MPH combined with Zerg/block if you need them...
Keep up the great work.

**Comment by UselessGuru** [comment]
> it seems to be some problem related to MPH

Could you please check the time stampf of all MPH stat files?
I suspect that some file will be older.
As soon as you remove this/these files the problem should go away. Can you confirm this?

**Comment by titvs** [comment]
Well, all of the stat files regarding MultiPoolHub were dated today, 31st May... 
Anyway, i did what you asked and deleted all of them related to MPH. Started MPM with MPH+Zerg but the problem persists with the "out of sync" message popping out.
Switched back to just Zerg + Blockmasters and no "out of sync" error...

Tomorrow i will try to do a full reset + benchmark and post my findigs.

**Comment by grantemsley** [comment]
I too have been seeing more out of sync messages, often with times of around 3000 minutes, far longer than I'd expect if a pool was just having intermittent issues.  I haven't looked into it yet though.

**Comment by UselessGuru** [comment]
My suggestion is that some pool drops support for some algo/coin. This will then leave orphaned stat info which triggers the warning.

If this is the root  cause, then we should change MPM so it just compares the time stamps for the newest stat per pool. And we should even automatically remove all orphaned stat files too.

**Comment by titvs** [comment]
Ok, today I’ve reset MPM (removed logs, stats and profit) and did a normal benchmark run using Mining Pool Hub and ZergPool. The out of sync message appeared a couple of times (10 min, 20 min...) during benchmark and after benchmark.
I’ve noticed again odd switching with it selecting the least profitable algo on the list... weird... On another run with just MPH it selected the 3rd most profitable algo on 4 consecutive cycles.
It’s important to mention that I have active the new Use Fastest Miner per Algo switch.

I’ve switched back to just ZergPool and Blockmasters and no issues arised... Are you guys sure that you didn’t do something to the MPH Module?

I’m going to take a careful look at the algos configured on the BAT file to make sure there isn’t any unsupported and make test runs without the Use Fastest Miner switch...

**Comment by UselessGuru** [comment]
> I’ve reset MPM (removed logs, stats and profit)

You can not delete stats AND profit. ???
There are 3 types of files in the 'Stats' folder:
*_Hashrate.txt -> benchmark files per miner & algo
*_Profit.txt -> accuracy per algo/coin & pool
profit.txt -> Used in summary for MPM is ... more profitable than ...

>  selected the 3rd most profitable algo

What do you mean be ...most profitable algo...?

MPM choses the BEST algo. See readme.
The BEST is determined by profit AND accuracy. MPM stores accuracy in the stat file. When you delete the stat file MPM needs to rebuild this value again (will be fintuned on every loop).

**Comment by titvs** [comment]
Sorry for my late reply. I’ve been testing out some of the latest commits but had to go to previous ones because of some problems already reported by others (launching amd miners on nvidia only rig, changed switch names at the config file but not at the bat file... these were some of the errors I got).
So I deleted everything and runned from scratch with MPH and ZergPool and I’m not seeing out of sync errors... well, not that long I was getting (200 min and above), so I guess the problem went away.
I’ve checked the enabled algos at the bat file and they are supported at the pools list so I really don’t know what happened...
You guys can close the issue but I’ll be on the lookout for similar behaviors it the future.
